atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleQueryFactory.java

```diff
@@ -346,7 +346,10 @@ public FullQuery getRowsColumnRangeCountsQuery(
     protected FullQuery getRowsColumnRangeSubQuery(
             byte[] row, long ts, BatchColumnRangeSelection columnRangeSelection) {
         String query = " /* GET_ROWS_COLUMN_RANGE (" + tableName + ") */ "
-                + "SELECT s.row_name, s.col_name, s.ts" + getValueSubselect("s", true)
+                + "SELECT "
+                + "/*+ NO_INDEX_FFS(@SEL$2 M@SEL$2(" + tableName + ".row_name, " + tableName + ".col_name, "
```

Comment:
should we change the syntax elsewhere in the query to give aliases to things so that you can reference them in hints? or is that not possible here?

Reply:
It sounds like in this case it actually is't possible to alias things.

Reply:
We are aliasing the tablename: hint syntax for NO_INDEX_FFS is /*+ NO_INDEX_FFS([optional query block] table index) */

The table is aliased to M - in SEL$2.

The weird thing is that Oracle doesn't accept the index PK_ + tablename here, and I had to use this weird notation I discovered by looking at the detailed plan generated in sqldeveloper.

And I'm glad I just revisited this, because there is a missing space after M@SEL2